### PR TITLE
Tests for Streamed Mode in SynchronizationContext

### LINKED ISSUE #223: Add Tests covering TransferMode.Streamed using SynchronizationContext
Per https://github.com/dotnet/wcf/pull/209/files#r35457169, we should add Tests covering using TransferMode.Streamed with SynchronizationContext and see if we would get deadlock.

### PR DESCRIPTION
**Pull request by shmao**
Added tests covering TransferMode.Streamed using SynchronizationContext. 

Fixes #223.

### CODE REVIEW

**Comment by shmao** [comment]
@roncain @mconnew @StephenBonikowsky

File: src/System.Private.ServiceModel/tests/Scenarios/Contract/Service/ServiceContractTests.cs [hunk 1]
```diff
@@ -159,6 +161,34 @@ public static void DefaultSettings_Echo_RoundTrips_String_Streamed_Async()
...
+    {
+        bool success = Task.Run(() =>
+        {
+            TestTypes.SingleThreadSynchronizationContext.Run(() =>
+            {
+                Task.Factory.StartNew(() => ServiceContractTests.DefaultSettings_Echo_RoundTrips_String_Streamed(), CancellationToken.None, TaskCreationOptions.None, TaskScheduler.FromCurrentSynchronizationContext()).Wait();
+            });
+        }).Wait(ScenarioTestHelpers.TestTimeout);
+        Assert.True(success, "Test Scenario: DefaultSettings_Echo_RoundTrips_String_Streamed_WithSingleThreadedSyncContext");
+    }
```
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Scenarios/Contract/Service/ServiceContractTests.cs]: Nit: I would add the word "timed-out" to the end of the assert message so someone reading the run results know what the Assert.True means when it fails.  Same for the other test.

**Comment by roncain** [comment]
LGTM, subject to a couple optional NIT fixes.
@mconnew would you mind taking a look to verify these tests look right for testing async streaming within a SynchronizationContext environment?

File: src/System.Private.ServiceModel/tests/Scenarios/Contract/Service/ServiceContractTests.cs (hunk 1 above)
Comment (shmao) [inline src/System.Private.ServiceModel/tests/Scenarios/Contract/Service/ServiceContractTests.cs]: Added "timed-out" to the end of the message. Thanks for reviewing the PR, Ron.

**Comment by iamjasonp** [comment]
LGTM.

**Comment by mconnew** [comment]
@roncain, looks correct to me.
LGTM :shipit: :+1:

**Comment by shmao** [comment]
Thank you all for reviewing the PR.

**Pull request merged**